File: idragonfly/MATLAB Codes/assembleMatrix.m
function [ MVN ] = assembleMatrix( m,MVNs,MVNs_12,MVNs_21 )
%Assemblle 4 sub-matrices 
%INPUT
% MVNs      diagonal
% MVNs_12   off-diagonal
% MVNs_21   off-diagonal
    MVN(   1 :  m,    1 :  m)=MVNs(   1:m,1:m,1);
    MVN(   1 :  m, (m+1):2*m)=MVNs_12(1:m,1:m);
    MVN((m+1):2*m,    1 :  m)=MVNs_21(1:m,1:m);
    MVN((m+1):2*m, (m+1):2*m)=MVNs(   1:m,1:m,2);
end

File: idragonfly/Python Codes/assembleMatrix.py
import jax.numpy as jnp

def assembleMatrix(m, MVNs, MVNs_12, MVNs_21):
    """
    Assemblle 4 sub-matrices 
    INPUT
    MVNs      diagonal
    MVNs_12   off-diagonal
    MVNs_21   off-diagonal
    """

    MVN = jnp.zeros((2*m, 2*m))

    MVN = MVN.at[:m, :m].set(MVNs[:m, :m, 0])
    MVN = MVN.at[:m, m:2*m].set(MVNs_12[:m, :m])
    MVN = MVN.at[m:2*m, :m].set(MVNs_21[:m, :m])
    MVN = MVN.at[m:2*m, m:2*m].set(MVNs[:m, :m, 1])

    return MVN